### PR TITLE
fix: use simple python image to build mkdocs

### PR DESCRIPTION
**Pull request by csantanapr**
Fixes #17311 

Instead of using specific container image from mkdocs-material, a cleaner approach is to match what the [.readthedocs.yaml](https://github.com/argoproj/argo-cd/blob/master/.readthedocs.yml) config uses a base image with python, then install the pip packages

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Comment by csantanapr** [comment]
The Python Software Foundation (PSF) will no longer support Python 3.7 on June 27th, 2023

We should try bump to 3.8 for the .readthedocs.yaml but that would need to be done in a different PR and do testing with .readthedocs build system

File: Makefile [hunk 1]
```diff
@@ -27,9 +27,8 @@ DOCKER_WORKDIR?=/go/src/github.com/argoproj/argo-cd
 
 ARGOCD_PROCFILE?=Procfile
 
-# Strict mode has been disabled in latest versions of mkdocs-material.
-# Thus pointing to the older image of mkdocs-material matching the version used by argo-cd.
-MKDOCS_DOCKER_IMAGE?=squidfunk/mkdocs-material:4.1.1
+# pointing to python 3.8 as the oldest version of python supported and compatible with argocd-docs mkdocs/mkdocs-material
+MKDOCS_DOCKER_IMAGE?=python:3.8-alpine
```
Comment (morey-tech) [inline Makefile:31]: Should we match the python version to [what's used in `.readthedocs`](https://github.com/argoproj/argo-cd/blob/2ac96a55badf673cf16acc4a959ac7b3e325e1cf/.readthedocs.yml#L11)? 
```suggestion
MKDOCS_DOCKER_IMAGE?=python:3.7-alpine
```

Comment (csantanapr) [reply Makefile:31]: See my comment
>The Python Software Foundation (PSF) will no longer support Python 3.7 on June 27th, 2023

We should try bump to 3.8 for the [.readthedocs.yaml](https://github.com/argoproj/argo-cd/blob/master/.readthedocs.yml) but that would need to be done in a different PR and do testing with .readthedocs build system

Comment (csantanapr) [reply Makefile:31]: let me check if the image `3.7-alpine` is availble

Comment (morey-tech) [reply Makefile:31]: Correction, I just saw the comment, should we update what's in the .readthedocs to 3.8 then?

Comment (csantanapr) [reply Makefile:31]: its looks like is available, I will update to 3.7, and then see when can we move everything to a newer version of python 3.8+

Comment (csantanapr) [reply Makefile]: 3.8 in .readthedocs should be a different issue/pr combination

**Review by morey-tech — APPROVED** [review]

**Review by crenshaw-dev — APPROVED** [review]
Thanks!

**Comment by csantanapr** [comment]
Thank you for merging @crenshaw-dev

**Pull request merged**